config.go

```diff
@@ -1136,6 +1143,28 @@ func loadConfig() (*config, []string, error) {
 		}
 	}
 
+	// Clamp GOMAXPROCS between 1 and 10x number of CPU threads
+	if cfg.GoMaxProcs < 1 {
+		cfg.GoMaxProcs = 1
+	}
+	if cfg.GoMaxProcs > runtime.NumCPU()*10 {
+		cfg.GoMaxProcs = runtime.NumCPU() * 10
+	}
+
+	// Set the configured value on the runtime.
+	runtime.GOMAXPROCS(cfg.GoMaxProcs)
+
+	// Clamp GOGC between 1 and 1000
+	if cfg.GoGC < 1 {
+		cfg.GoGC = 1
+	}
+	if cfg.GoGC > 1000 {
```

Comment:
FWIW this can already be set as an env variable: https://github.com/btcsuite/btcd/blob/718d268a8cf55af9b329bac9e171009352fbc10a/btcd.go#L331-L338